### PR TITLE
Global collectd update

### PR DESCRIPTION
**Pull request by cznewt**

### CODE REVIEW

File: collectd/files/backend/carbon.conf [hunk 1]
```diff
@@ -1,7 +1,8 @@
-{%- from "linux/map.jinja" import system with context %}
 {%- set backend = salt['pillar.get']('collectd:client:backend:'+backend_name) %}
-
-LoadPlugin write_graphite
+{%- from "linux/map.jinja" import system with context %}
```
Comment (fpytloun) [inline collectd/files/backend/carbon.conf]: Maybe it's time for getting rid of this weirdness and sick dependency on linux:

```
Prefix "{{ system.get('cluster', 'prd') }}_{{ system.get('environment', 'prd') }}."
```

and make it configurable as a part of collectd metadata (+ fix legacy deployments).

File: collectd/meta/collectd.yml [hunk 2]
```diff
@@ -40,5 +39,5 @@ plugin:
         values:
         - 1.3.6.1.2.1.31.1.1.1.7
         - 1.3.6.1.2.1.31.1.1.1.11
-    host: {{ pillar.external.network_device }}
-{%- endif %}
+    host: {{ pillar.external.network_device|json }}
```
Comment (fpytloun) [inline collectd/meta/collectd.yml]: just to be sure, shouldn't this be yaml?

**Comment by elemoine** [comment]
Quick question: you finally decided to create this PR against the master branch? I guess that's because it's preparation code not directly related to the introduction of the new monitoring stack. Right?

File: collectd/files/collectd_logfile.conf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<LoadPlugin logfile>
```
Comment (simonpasquier) [inline collectd/files/collectd_logfile.conf:1]: This is unrelated to this PR but
1/ this doesn't seem to be used anywhere by default so collectd isn't logging anything.
2/ the configuration files should be correctly ordered, eg you want the logging plugin to be loaded first to be able to troubleshoot collectd configuration problems [0].

[0] https://github.com/voxpupuli/puppet-collectd/pull/166#issuecomment-50591413

**Review by simonpasquier — COMMENTED** [review]
LGTM

**Review by simonpasquier — APPROVED** [review]

**Review by elemoine — APPROVED** [review]

**Comment by cznewt** [comment]
Yes this is merged against master as it is major cleanup compatible with what it does now and making it compatible with stacklight as well. it will be best merged with linux, nginx, and other services in same time to upgrade.

**Pull request merged**